docs/pages/x/api/date-pickers/date-picker.json

```diff
@@ -11,6 +11,10 @@
     "className": { "type": { "name": "string" } },
     "clearable": { "type": { "name": "bool" } },
     "clearText": { "type": { "name": "node" }, "default": "'Clear'" },
+    "closeOnSelect": {
+      "type": { "name": "bool" },
+      "default": "`true` for Desktop, `false` for Mobile (based on the chosen wrapper and `desktopModeMediaQuery` prop)."
+    },
```

Comment:
Do we have more context on this change? If we follow https://mui.com/material-ui/guides/api/#prop-naming. I would expect the prop to be called `disableCloseOnSelect` since the desktop is more important than mobile (in terms of usage frequency).

Reply:
For me `disableCloseOnSelect` is harder to understand than `closeOnSelect` and the weight for of the desktop vs mobile does not counter this complexity.

While working on the code related to this prop, I had at least 3 times to switch the behavior because I wrote it in the wrong order.
And the doc was inversed (it was written for `closeOnSelect` not `disableCloseOnSelect`)

I don't think `disableXXX` with a dynamic default value that can be `true` or `false` is a good practice. It's not readable when you don't know the codebase well enough.

Reply:
> I don't think disableXXX with a **dynamic** default value that can be true or false is a good practice. 

Alright, deal, sounds like we have a new rule for prop names with dynamic default. I would be in favor of adding this to https://mui.com/material-ui/guides/api/#prop-naming